hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -128,12 +128,7 @@ public JdbcEnvironment initiateService(Map<String, Object> configurationValues,
 					explicitDatabaseMinorVersion,
 					explicitDatabaseVersion);
 		}
-		else if ( explicitDialectConfiguration(
-				configurationValues,
-				explicitDatabaseName,
-				explicitDatabaseMajorVersion,
-				explicitDatabaseMinorVersion,
-				explicitDatabaseVersion) ) {
+		else if ( isNotEmpty(explicitDatabaseName) || isNotNullAndNotEmpty( configurationValues.get(DIALECT) ) ) {
```

Comment:
I think it would be better to keep `explicitDialectConfiguration` and simply remove the version checks from there.

Reply:
```suggestion
		else if ( explicitDialectConfiguration( explicitDatabaseName, configurationValues ) ) {
```

Reply:
Well I tried adding suggestion, but deleted lines are not supported :)

```java
	private static boolean explicitDialectConfiguration(
			String explicitDatabaseName,
			Map<String, Object> configurationValues) {
		return isNotEmpty(explicitDatabaseName) || isNotNullAndNotEmpty( configurationValues.get(DIALECT ) );
	}
```

Reply:
Applied, thanks.